README.md

```diff
@@ -3,3 +3,23 @@
 [![Build Status](https://travis-ci.org/jzych/Hardcore-Dining-Philosophers.svg?branch=master)](https://travis-ci.org/jzych/Hardcore-Dining-Philosophers)
 
 Extended version of Dining Philosopher problem
+
+## 1.0 Architecture
+
+![Architecture diagram](doc/main_diagram.png)
+
+*CTable* - contains two vectors of philosophers and forks.
+It is responsible for init and end.
+
+CPhilosooher - A thread with life-cycle of philosopher function.
+It contains reference to its forks ie. "F0-F1".
+This class writes its activity to CLog.
+It asks CWaiter if can eat at specified (current) time.
+
+*CFork* - fork representation.
+
+*CLog* - log (queue) with philosophers activities.
+
+*CoutputConsole* - class reponsible for display log on second console.
```

Comment:
CoutputConsole - moze to byc funkcja member Log, ktora odpala w konstruktorze jako osobny thread, wtedy tez ~Log bylby odpowiedzialny na jakie ubicie i join() w swoim destruktorze.

Reply:
Myślę, że lepiej będzie zostawić tu klasę tak jak jest. Wydaje mi się, że może być nieco bardziej skomplikowana niż tylko jedna funkcja. Jakby co to poprawię diagram.
Poza tym zasada "pojedynczej odpowiedzialności".

Reply:
Hmm ok, chociaz wydawalo mi sie ze logowanie zawiera w sobie wyrzucanie na output jako pojedyncza odpowiedzialnosc